src/core/containers/runtime/uv_provider.py

```diff
@@ -0,0 +1,183 @@
+"""Providers for launching Hugging Face Spaces via ``uv run``."""
+
+from __future__ import annotations
+
+import os
+import socket
+import subprocess
+import time
+from dataclasses import dataclass, field
+from typing import Dict, Optional
+
+import requests
+
+from .providers import ContainerProvider
+
+
+def _poll_health(health_url: str, timeout_s: float) -> None:
+    """Poll a health endpoint until it returns HTTP 200 or times out."""
+
+    deadline = time.time() + timeout_s
+    while time.time() < deadline:
+        try:
+            response = requests.get(health_url, timeout=2.0)
+            if response.status_code == 200:
+                return
+        except requests.RequestException:
+            pass
+
+        time.sleep(0.5)
+
+    raise TimeoutError(
+        f"Server did not become ready within {timeout_s:.1f} seconds"
+    )
+
+
+def _create_uv_command(
+    repo_id: str,
+    host: str,
+    port: int,
+    reload: bool,
+    project_url: Optional[str] = None,
+) -> list[str]:
+    command = [
+        "uv",
+        "run",
+        "--project",
+        project_url or f"git+https://huggingface.co/spaces/{repo_id}",
+        "--",
+        "server",
+        "--host",
+        host,
+        "--port",
+        str(port),
+    ]
+    if reload:
+        command.append("--reload")
+    return command
+
+
+@dataclass
+class UVProvider(ContainerProvider):
+    """ContainerProvider implementation backed by ``uv run``."""
+
+    repo_id: str
+    host: str = "0.0.0.0"
+    port: Optional[int] = None
+    reload: bool = False
+    project_url: Optional[str] = None
+    connect_host: Optional[str] = None
+    extra_env: Optional[Dict[str, str]] = None
+    context_timeout_s: float = 60.0
+
+    _process: subprocess.Popen | None = field(init=False, default=None)
+    _base_url: str | None = field(init=False, default=None)
+
+    def start_container(
+        self,
+        image: str,
+        port: Optional[int] = None,
+        env_vars: Optional[Dict[str, str]] = None,
+        **_: Dict[str, str],
+    ) -> str:
+        if self._process is not None and self._process.poll() is None:
+            raise RuntimeError("UVProvider is already running")
+
+        self.repo_id = image or self.repo_id
+
+        bind_port = port or self.port or self._find_free_port()
+
+        command = _create_uv_command(
```

Comment:
In `LocalDockerProvider.__init__`, we are checking if Docker is installed ([here](https://github.com/meta-pytorch/OpenEnv/blob/01a583538df5a2909cef4f146e9309fa9fd0e900/src/core/containers/runtime/providers.py#L106)). I think it'd be good to do the same for uv. Some like 

```py
         try:
            subprocess.run(
                ["uv", "self", "version"],
                check=True,
                capture_output=True,
                timeout=1,
            )
        except (subprocess.CalledProcessError, FileNotFoundError, subprocess.TimeoutExpired):
            raise RuntimeError(
                "uv is not available. Please install it to use UVProvider."
            )
```

Doing so we would fail early instead of failing when trying to start the project.